db/model/Image.ts

```diff
@@ -297,3 +304,22 @@ export async function insertImageObject(
     const [id] = await knex.table("images").insert(image)
     return id
 }
+
+export async function fetchImagesFromDriveAndSyncToS3(
+    knex: KnexReadWriteTransaction,
+    filenames: string[] = []
+): Promise<Image[]> {
+    if (!filenames.length) return []
+
+    const images = await imageStore
+        .fetchImageMetadata(filenames)
+        .then((obj) => Object.values(obj))
+        .then(excludeUndefined)
+
+    return Promise.all(images.map((i) => Image.syncImage(knex, i)))
+        .then(excludeUndefined)
+        .catch((e) => {
+            console.error(`Error syncing images to S3`, e)
+            return []
+        })
```

Comment:
If we do this a lot less often now should we then maybe take care of the errors a bit better? It might be worth capturing the errors so we can report them up the call chain and back to the client so we can warn the user if some images didn't sync. What do you think?

Reply:
Sounds good. I've added errors in the right places so that users will get a JSON 500 response if something bad happens on fetch from gdrive or sync.

Far from ideal, but programming is a process 🙂